components/server/src/workspace/workspace-service.ts

```diff
@@ -610,6 +657,7 @@ export class WorkspaceService {
 
         // at this point we're about to actually start a new workspace
         const result = await this.workspaceStarter.startWorkspace(ctx, workspace, user, await projectPromise, options);
```

Comment:
IMO we should call it on `createWorkspace`, too, to prevent dead DB entries of workspaces that have never been started (based on `workspace.creationTime`).